biz.aQute.bndlib/src/aQute/bnd/build/Project.java

```diff
@@ -2112,6 +2112,15 @@ public void test(File reportDir, List<String> tests) throws Exception {
 			IO.delete(reportDir);
 			tester.setReportDir(reportDir);
 		}
+		if (tests == null) {
+			// check system properties: gradle
+			String singleTest = System.getProperty("integTest.single");
```

Comment:
This is not the right property name. See <https://docs.gradle.org/4.0/userguide/java_plugin.html#sec:single_test_execution_via_system_properties>. The property name is based upon the gradle task name. And you cannot really know the gradle task name here in bndlib. The standard task name for the BndPlugin is `testOSGi`.

Furthermore, value of the property in gradle is a globbing pattern.

Finally, this is the wrong place to put maven and gradle related support. Such support needs to go in the appropriate plugins.